packages/db-mongodb/src/queries/sanitizeQueryValue.ts

```diff
@@ -62,7 +62,7 @@ export const sanitizeQueryValue = ({
     formattedValue = Number(val)
   }
 
-  if (field.type === 'date' && typeof val === 'string') {
+  if (field.type === 'date' && typeof val === 'string' && operator !== 'exists') {
```

Comment:
It seems the problem is not on the `operator` but on the value instead, any boolean string should be cast to boolean type.

Reply:
`Boolean` state is used when `field.type` is `checkbox` and when operator is `exists`. It is work when `field.type` is `checkbox`.

But not when operator is `exists` and `field.type` is `date`. Because conditional logic you reviewed is parsing to Date early and return `undefined` when value is not available to parse Date object.

So I change to skip conditional logic when operator is `exists`. I thought about moving the conditional logic for `exists` back to before Date parsing, but that would cause bugs, so I went with this approach.